pycbc/events/single.py

```diff
@@ -141,13 +167,27 @@ def verify_args(args, parser, ifos):
 
     @classmethod
     def from_cli(cls, args, ifo):
+        # Allow None inputs
+        stat_files = args.statistic_files or []
+        stat_keywords = args.statistic_keywords or []
+
+        # flatten the list of lists of filenames to a single list
+        # (may be empty)
+        stat_files = sum(stat_files, [])
+
+        kwargs = stat.parse_statistic_keywords_opt(stat_keywords)
         return cls(
            ifo, newsnr_threshold=args.single_newsnr_threshold[ifo],
            reduced_chisq_threshold=args.single_reduced_chisq_threshold[ifo],
            duration_threshold=args.single_duration_threshold[ifo],
            fixed_ifar=args.single_fixed_ifar,
+           maximum_ifar=args.single_maximum_ifar[ifo],
            fit_file=args.single_fit_file,
-           sngl_ifar_est_dist=args.sngl_ifar_est_dist[ifo]
+           sngl_ifar_est_dist=args.sngl_ifar_est_dist[ifo],
+           statistic=args.ranking_statistic,
```

Comment:
I am not sure I understand why `--ranking-statistic` is relevant to the `LiveSingle` class. I probably missed or forgot some conversation…

Reply:
This is so that it can be passed into the stat_calculator property, which is used later (line 229)